MonoidalCategories/gap/SymmetricClosedMonoidalCategoriesDerivedMethods.gi

```diff
@@ -907,3 +907,27 @@ AddDerivationToCAP( MonoidalPostComposeMorphismWithGivenObjects,
     
 end : CategoryFilter := IsSymmetricClosedMonoidalCategory and IsStrictMonoidalCategory,
       Description := "MonoidalPostComposeMorphismWithGivenObjects using evaluation, and tensor hom adjunction" );
+
+##
+AddDerivationToCAP( IsomorphismFromDualToInternalHom,
+        
+  function( cat, a )
+    
+    return PreCompose( cat,
+                   RightUnitorInverseWithGivenTensorProduct( cat, a ),
+                   MorphismFromTensorProductToInternalHom( cat, a, TensorUnit( cat ) ) );
+    
+end : CategoryFilter := IsSymmetricClosedMonoidalCategory,
+      Description := "IsomorphismFromDualToInternalHom using MorphismFromTensorProductToInternalHom" );
+
+##
+AddDerivationToCAP( IsomorphismFromDualToInternalHomWithGivenObjects,
+        
+  function( cat, source, a, range )
+    
+    return PreCompose( cat,
+                   RightUnitorInverseWithGivenTensorProduct( cat, a, source ),
+                   MorphismFromTensorProductToInternalHomWithGivenObjects( cat, source, a, TensorUnit( cat ), range ) );
```

Comment:
We have `source = a^v` but `MorphismFromTensorProductToInternalHomWithGivenObjects` expects the second argument to be `a^v x 1`.

Reply:
Right.